runelite-api/src/main/java/net/runelite/api/events/EventBus.java

```diff
@@ -0,0 +1,248 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.api.events;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import java.lang.invoke.CallSite;
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.ThreadSafe;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+@ThreadSafe
+public class EventBus
+{
+	@FunctionalInterface
+	public interface SubscriberMethod
+	{
+		void invoke(Object event);
+	}
+
+	@Value
+	private static class Subscriber
+	{
+		private final Object object;
+		private final Method method;
+		@EqualsAndHashCode.Exclude
+		private final SubscriberMethod lamda;
+
+		void invoke(final Object arg) throws Exception
+		{
+			if (lamda != null)
+			{
+				lamda.invoke(arg);
+			}
+			else
+			{
+				method.invoke(object, arg);
+			}
+		}
+	}
+
+	private final Consumer<Throwable> exceptionHandler;
+	private ImmutableMultimap<Class, Subscriber> subscribers = ImmutableMultimap.of();
+
+	/**
+	 * Instantiates EventBus with default exception handler
+	 */
+	public EventBus()
+	{
+		this((e) -> log.warn("Uncaught exception in event subscriber", e));
```

Comment:
If this is the default exception handler we don't need the one in the runelite module anymore. I think it was only there because the default guava one ate exceptions?

Reply:
Alright removed the runelite module one, and yes